mindmeld/converter/dialogflowconverter.py

```diff
@@ -0,0 +1,316 @@
+
+        datastore = json.load(source_en)
+        allText = []
+
+        for usersay in datastore:
+            sentence = ""
+            for texts in usersay["data"]:
+                df_text = texts["text"]
+                if "meta" in texts and texts["meta"] != "@sys.ignore":
+                    df_meta = texts["meta"]
+
+                    if re.match("(@sys.).+", df_meta):  # if text is a dialogflow sys entity
+                        if df_meta in DialogFlowConverter.sys_entity_map:
+                            mm_meta = DialogFlowConverter.sys_entity_map[df_meta]
+                        else:
+                            mm_meta = "[DNE: {sysEntity}]".format(sysEntity=df_meta[1:])
+                            logger.info("Unfortunately mindmeld does not currently support"
+                                        "%s as a sys entity."
+                                        "Please create an entity for this.", df_meta[1:])
+
+                        part = "{" + df_text + "|" + mm_meta + "}"
+                    else:
+                        part = "{" + df_text + "|" + df_meta[1:] + "}"
+                else:
+                    part = df_text
+
+                sentence += part
+            allText.append(sentence)
+
+        train, test = train_test_split(allText, test_size=0.2)
+
+        target_test.write("\n".join(test))
+        target_train.write("\n".join(train))
+
+        source_en.close()
+        target_test.close()
+        target_train.close()
+
+    def _get_file_names(self, level):
+        """ Gets the names of the entities from DialogFlow as a dictionary.
+        ex. if we had the following files in our entities directory:
+            ["test.json", "test_entries_en.json", "test_entries_de.json"]
+        return:
+            {'test': {'en': 'test_entries_en', 'de': 'test_entries_de'}} """
+
+        directory = os.path.join(self.dialogflow_project_directory, level)
+        files = os.listdir(directory)
+
+        w = {"entities": "entries", "intents": "usersays"}
+        p = r".+(?<=(_" + w[level] + "_))(.*)(?=(.json))"
+
+        info = {}
+        for name in files:
+            match = re.match(p, name)
+
+            if match:
+                isbase = False
+                base = name[:match.start(1)]
+                language = match.group(2)
+            else:
+                isbase = True
+                base = name[:-5]
+
+            if base not in info:
+                info[base] = {}
+
+            if not isbase:
+                info[base][language] = name[:-5]
+
+        return info
+
+    def create_training_data(self):
+        entities = self._get_file_names("entities")
+        self._create_entities_directories(entities)
+
+        intents = self._get_file_names("intents")
+        self._create_intents_directories(intents)
+
+    # =========================
+    # create init
+    # =========================
+
+    @staticmethod
```

Comment:
@barret-p can we reuse/refactor these methods into the parent class?
```suggestion
    @staticmethod
```

Reply:
Does this still need to be addressed?